plugins/uefi-capsule/fu-uefi-device.c

```diff
@@ -567,11 +632,15 @@ fu_uefi_device_write_firmware (FuDevice *device,
 	fw = fu_firmware_get_bytes (firmware, error);
 	if (fw == NULL)
 		return FALSE;
-
+	/* check if we want a capsule update on-disk */
+	on_disk = fu_check_capsuleupdate_ondisk(device, error);
 	/* save the blob to the ESP */
 	directory = fu_uefi_get_esp_path_for_os (device, esp_path);
 	basename = g_strdup_printf ("fwupd-%s.cap", self->fw_class);
-	fn = g_build_filename (directory, "fw", basename, NULL);
+	if (on_disk)
+		fn = g_build_filename (directory, "EFI/UpdateCapsule", basename, NULL);
```

Comment:
is this path actually right?  won't it end up in a distro specific path (for example /efi/ubuntu/efi/updatecapsule)?

Reply:
Yes I think it is. The UEFI spec says:
"The directory \EFI\UpdateCapsule is checked for capsules only within the EFI system partition on the device specified in the active boot option determine by reference to BootNext variable or BootOrdervariable processing. The active Boot Variable is the variable with highest priority BootNext or within BootOrder that refers to a device found to be present. Boot variables in BootOrder but referring to devices not present are ignored when determining active boot variable"

So your BootNext or Boot#### variables define the directory you should look into for EFI/UpdateCapsule/<capsules>

Reply:
I managed to set this up on an rpi4 for testing. I misunderstood your original point. I assumed "/efi/ubuntu' was part of were the OS mounted the ESP.  This is indeed wrong, I'll fix it